tests/modules/hooks.c

```diff
@@ -287,9 +289,51 @@ void keyInfoCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub, void
     }
 
     RedisModuleKeyInfoV1 *ei = data;
-    RedisModuleString *keyname = RedisModule_CreateStringPrintf(ctx, "key-info-%s", ei->key);
-    LogStringEvent(ctx, RedisModule_StringPtrLen(keyname, NULL), ei->key);
-    RedisModule_FreeString(ctx, keyname);
+    RedisModuleString *key = ei->key;
+    const char* keyname = RedisModule_StringPtrLen(key, NULL);
+    RedisModuleString *event_keyname = RedisModule_CreateStringPrintf(ctx, "key-info-%s", keyname);
+    LogStringEvent(ctx, RedisModule_StringPtrLen(event_keyname, NULL), keyname);
+    RedisModule_FreeString(ctx, event_keyname);
+
+    /* Open Key to check key exists */
+    RedisModuleKey *kp = RedisModule_OpenKey(ctx, key, REDISMODULE_READ);
+    if(kp != NULL) {
```

Comment:
i guess we wanna assert that the RM_OpenKey succeeded instead of ignore it?

Reply:
If `kp` is NULL, the test will fail. So we maybe needn't to assert it?

Reply:
so maybe drop the `if`, and just let it crash?
i think it looks confusing to see the `else` isn't handled. so even if assert isn't needed, maybe it makes things clearer.